src/Navbar.js

```diff
@@ -86,12 +105,22 @@ const Navbar = React.createClass({
       <ComponentClass {...this.props} className={classNames(this.props.className, classes)}>
         <div className={this.props.fluid ? 'container-fluid' : 'container'}>
           {(this.props.brand || this.props.toggleButton || this.props.toggleNavKey != null) ? this.renderHeader() : null}
```

Comment:
Isn't this going to re-render the header toggle bits even when a `NavBrand` is specified?